tgui/packages/tgui/interfaces/ReagentGrinder.js

```diff
@@ -0,0 +1,166 @@
+import { useBackend } from '../backend';
+import { Box, Button, Section, Table, Flex, Icon, Dimmer } from '../components';
+import { Window } from '../layouts';
+import { BeakerContents } from '../interfaces/common/BeakerContents';
+
+export const ReagentGrinder = (props, context) => {
+  return (
+    <Window resizable>
+      <Window.Content scrollable display="flex" className="Layout__content--flexColumn">
+        <Operating/>
+        <GrinderControls/>
+        <GrinderContents/>
+        <GrinderReagents/>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const Operating = (props, context) => {
+  const { data } = useBackend(context);
+  const {
+          operating,
+          name
+        } = data;
+
+  if (operating) {
+    return (
+      <Dimmer>
+        <Flex mb="30px">
+          <Flex.Item
+            bold
+            color="silver"
+            textAlign="center">
+            <Icon
+              name="spinner"
+              spin
+              size={4}
+              mb="15px"
+            /><br />
+            The {name} is processing...
+          </Flex.Item>
+        </Flex>
+      </Dimmer>
+    );
+  }
+};
+
+const GrinderControls = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+          inactive,
+        } = data;
+
+  return (
+    <Section title="Controls">
+      <Flex>
+        <Flex.Item width="50%" mr="3px">
+          <Button
+            fluid
+            textAlign="center"
+            icon="mortar-pestle"
+            disabled={inactive}
+            tooltip={inactive ? "There are no contents" : "Grind the contents"}
+            tooltipPosition="bottom"
+            content="Grind"
+            onClick={() => act('grind')}
+          />
+        </Flex.Item>
+        <Flex.Item width="50%">
+          <Button
+            fluid
+            textAlign="center"
+            icon="blender"
+            disabled={inactive}
+            tooltip={inactive ? "There are no contents" : "Juice the contents"}
+            tooltipPosition="bottom"
+            content="Juice"
+            onClick={() => act('juice')}
+          />
+        </Flex.Item>
+      </Flex>
+    </Section>
+  );
+}
+
+const GrinderContents = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    contents,
+    limit,
+    count,
+    inactive
+  } = data;
+
+  return (
+    <Section
+    title="Contents"
+    flexGrow={1}
+    buttons={(
```

Comment:
```suggestion
        title="Contents"
        flexGrow={1}
        buttons={(
```
These Section props should be indented over once to the right.